src/platform/ESP32/ConfigurationManagerImpl.cpp

```diff
@@ -262,19 +265,30 @@ CHIP_ERROR ConfigurationManagerImpl::StoreCountryCode(const char * code, size_t
     return GenericConfigurationManagerImpl<ESP32Config>::StoreCountryCode(code, codeLen);
 }
 
+#if CHIP_DEVICE_CONFIG_ENABLE_ETHERNET
+CHIP_ERROR ConfigurationManagerImpl::GetPrimaryMACAddress(MutableByteSpan buf)
+{
+    if (buf.size() != ConfigurationManager::kPrimaryMACAddressLength)
+        return CHIP_ERROR_INVALID_ARGUMENT;
+
+    return MapConfigError(esp_read_mac(buf.data(), ESP_MAC_ETH));
```

Comment:
After reading the mac address, we should reduce the span size to `ConfigurationManager::kPrimaryMACAddressLength`.

Reply:
Added these changes @shubhamdp